content/blog/webr-0-3-1/index.md

```diff
@@ -0,0 +1,368 @@
+
+In this interactive example, webR is configured to automatically install WebAssembly packages. Click "Run code" to download the packages listed in the R script.
+
+<div class="highlight">
+
+{{< webr-editor code=`# Explicitly install wasm packages\ninstall.packages("cli")\n\n# Automatically install wasm packages\nlibrary(vctrs)\nrequire(jsonlite)\n\n# Confirm the packages installed successfully\nrownames(installed.packages())` width=504 height=311.472 >}}
+
+</div>
+
+See the [documentation](https://docs.r-wasm.org/webr/latest/packages.html) for more details on how to control this behaviour in your own webR-powered applications, including optionally showing an interactive download menu to the user.
+
+### Error handling and reporting
+
+Improvements have been made to how webR raises R conditions as JavaScript exceptions. Exceptions now include the offending source R call in the error message text, better matching what is shown in a traditional R console.
+
+``` javascript
+await webR.evalR("sin('abc')");
+```
+
+<div class="output error">
+
+    Uncaught Error: Error in sin("abc"): non-numeric argument to mathematical function
+
+</div>
+
+Conditions raised when invoking function objects are now also re-thrown as JavaScript exceptions, rather than a generic `UnwindProtectException` error. Compare the error messages shown below from the previous and latest versions of webR to see the useful context added by this change.
+
+``` javascript
+// webR 0.2.2
+const do_calc = await webR.evalR(`function (n){ rnorm(n) }`)
+do_calc(-10)
+```
+
+<div class="output error">
+
+    Uncaught (in promise) UnwindProtectException: A non-local transfer of control occured during evaluation
+
+</div>
+
+``` javascript
+// webR 0.3.1
+const do_calc = await webR.evalR(`function (n){ rnorm(n) }`)
+do_calc(-10)
+```
+
+<div class="output error">
+
+    Uncaught (in promise) Error: Error in rnorm(n): invalid arguments
+
+</div>
+
+Some base R features can be problematic when running R under WebAssembly. For example, in the constrained WebAssembly sandbox the base R function [`system()`](https://rdrr.io/r/base/system.html) does not work. The latest release of webR now handles these cases more consistently, raising R [`stop()`](https://rdrr.io/r/base/stop.html) conditions rather than incorrectly returning an empty result.
+
+``` r
+# webR 0.3.1
+system()
+```
+
+<div class="output error">
+
+    Error in webr_hook_system(command) : 
+      The "system()" function is unsupported under Emscripten.
+
+</div>
+
+## Capturing HTML canvas graphics output
+
+The [`captureR()`](https://docs.r-wasm.org/webr/latest/evaluating.html#evaluating-r-code-and-capturing-output-with-capturer) function is designed to capture output generated when evaluating R code. With this release, plots drawn using webR's HTML canvas graphics device, [`webr::canvas()`](https://rdrr.io/pkg/webr/man/canvas.html), are also captured and returned by default.
```

Comment:
It might be interesting to mention the existing capture of errors and conditions, especially since the section just before is about errors.